virttest/utils_test/qemu/__init__.py

```diff
@@ -698,3 +699,306 @@ def get_all_memorys(self, vm):
         session = self.get_session(vm)
         output = session.cmd_output_safe(cmd, timeout=90)
         return set([_ for _ in output.splitlines() if _])
+
+
+class UsbDevTest(object):
+
+    """
+    Provide basic test for usb devices
+    """
+
+    def __init__(self, test, params, env):
+        self.params = params
+        self.test = test
+        self.env = env
+        self.vm = env.get_vm(params["main_vm"])
+
+    def check_usb_dev_monitor(self, usb_type):
+        """
+        check if the usb device exist in monitor
+        """
+        logging.info("Verify USB device in monitor.")
+        o = self.vm.monitor.info("usb")
+        devnums =[]
+        devs =[]
+        for line in o.splitlines():
+            if "testdev" in line:
+                for i in range(2):
+                    num = re.search("\d+(\.\d)?", line.split(",")[i]).group(0)
+                    devnums.append(num)
+                devs.append(devnums)
+        if devs:
+            return devs
+        else:
+            raise exceptions.TestFail("Could not find '%s' device, monitor "
+                                      "returns: \n%s" % (usb_type, o))
+
+    def check_usb_dev(self, usb_type):
+        """
+        check if the usb device exist in vm
+        """
+        if usb_type == 'usb-storage' or usb_type == 'usb-bot':
+            self.add_udisk()
+        timeout = float(self.params.get("login_timeout", 600))
+        session = self.vm.wait_for_login(timeout=timeout)
+        devs = self.check_usb_dev_monitor(usb_type)
+        if devs:
+            logging.info("The %s device can find in the monitor." % usb_type)
+        exist = self.check_usb_dev_guest(usb_type, session, devs)
+        if exist:
+            logging.info("The %s device was found in guest" % usb_type)
+        else:
+            raise exceptions.TestFail("Could not find '%s' in guest" % usb_type)
+
+    def check_usb_dev_guest(self, usb_type, session, devs):
+        """
+        check if the usb device exist in guest
+        """
+        logging.info("Verify %s device in guest." % usb_type)
+        cmd ="lsusb.py"
+        output = session.cmd_output(cmd)
+        print output
+        for devnum in devs:
+            for i in range(1,3):
+                key = "%s-%s" % (int(devnum[0][0])+i, devnum[1])
+                if key not in output:
+                    continue
+                else:
+                    return True
```

Comment:
If you want to check all the specified usb devices together, you should not return true since you only confirmed one device. If you just want to check one specified device,  you do not need go through this iteration at all.